sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -298,7 +298,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
                         case 'C':
                             handleClose(buffer, channel);
                             return;
-                        case 'X': // Terminate
+                        case 'X': // Terminate (called when jdbc connection is closed)
+                            closeSession();
+                            Messages.sendCloseComplete(channel);
```

Comment:
Looking at https://www.postgresql.org/docs/current/static/protocol-flow.html  Terminate there is no mentioning of the server having to send a closeComplete message?